web/api/app/Controller/AppController.php

```diff
@@ -58,14 +58,63 @@ class AppController extends Controller {
 	// Also checking to do this only if ZM_OPT_USE_AUTH is on
 	public function beforeFilter() {
 		$this->loadModel('Config');
+		
+        	$options = array('conditions' => array('Config.' . $this->Config->primaryKey => 'ZM_OPT_USE_API'));
+	 	$config = $this->Config->find('first', $options);
+        	$zmOptApi = $config['Config']['Value'];
+
+		if ($zmOptApi !='1')
+		{
+        		 throw new UnauthorizedException(__('API Disabled'));
+        	 	return; 
+		}
+		
         	$options = array('conditions' => array('Config.' . $this->Config->primaryKey => 'ZM_OPT_USE_AUTH'));
 	 	$config = $this->Config->find('first', $options);
         	$zmOptAuth = $config['Config']['Value'];
         	if (!$this->Session->Read('user.Username') && ($zmOptAuth=='1'))
         	{       
-        		 throw new NotFoundException(__('Not Authenticated'));
+        		 throw new UnauthorizedException(__('Not Authenticated'));
         	 	return; 
         	}     
+		// #1155 fixes
```

Comment:
Something like //User role check if you think it would help someone reading the code understand it's purpose, but for me doesn't need to talk about a change. That is all covered by commits and PR in github.